### PR TITLE
LPS-145457 Provide ways to determine if a Schema should be batch importable/exportable

### PR DESCRIPTION
**Pull request by ccorreagg**
This PR contains the code to determine programatically if a schema should be batch importable/exportable.

The following features has been included:

- Two new methods in the `VulcanBatchEngineTaskItemDelegate` interface. Those methods have a default implementation:

```
	public default Set<String> getAvailableCreateStrategies() {
		return null;
	}

	public default Set<String> getAvailableUpdateStrategies() {
		return null;
	}
```
**NOTE:** Each API Resource implementation will implement those methods without the `@Override` annotation **to avoid including changes that breaks previous versions.**

For example, for API Resource implementations that can manage any type of create and update strategies, the implementation would be the following:

```
	public Set<String> getAvailableCreateStrategies() {
		return SetUtil.fromArray("UPSERT", "INSERT");
	}

	public Set<String> getAvailableUpdateStrategies() {
		return SetUtil.fromArray("PARTIAL_UPDATE", "UPDATE");
	}
```
With this change, it will be possible to determine the available strategies for each Schema.

- Two new properties to determine if the Schema can be included in the Import and Export processes. That means the Schema is batch importable / batch exportable (because it contains a proper implementation of the `create` and `read` methods of the `VulcanBatchEngineTaskItemDelegate` interface).

```
batch.planner.import.enabled=true
batch.planner.export.enabled=true
```

### CODE REVIEW

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:test:sf

**Comment by ccorreagg** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java [hunk 1]
```diff
@@ -108,6 +108,10 @@ public synchronized List<ServiceRegistration<?>> deploy(
 						).put(
 							"batch.engine.task.item.delegate.name",
 							objectDefinition.getShortName()
+						).put(
+							"batch.planner.export.enabled", "true"
+						).put(
+							"batch.planner.import.enabled", "true"
```
Comment (LuismiBarcos) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java:114]: Thanks for remembering to put this in Objects 😁

**Review by LuismiBarcos — APPROVED** [review]
LGTM :) Let's see what Brain says about the method names. I would not change any of them. Really nice work! 👏 👏 👏

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by ccorreagg** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by ccorreagg** [comment]
PR forwarded manually: https://github.com/brianchandotcom/liferay-portal/pull/118512

Closing PR...

**Pull request closed without merging**